shared/src/main/scala/tastyquery/reader/TastyUnpickler.scala

```diff
@@ -12,6 +13,11 @@ import scala.collection.mutable
 
 object TastyUnpickler {
 
+  private def readerFromBytes(bytes: IArray[Byte]): TastyReader = {
+    import unsafe.unlock.given unsafe.CanObserveIArray[Byte]
+    new TastyReader(unsafe.asArray(bytes))
```

Comment:
This `CanObserveIArray` thing is completely overkill if it's used exactly once, in a private method.

Use
```suggestion
  new TastyReader(bytes.asInstanceOf[Array[Byte]])
```
instead.

In fact, it is not actually safe to this anyway. This is called from the public constructor of `TastyUnpickler`. So we *must* make a copy to be safe.

Reply:
tasty unpickler does not mutate the array, its only unfortunate that it uses array in its api

Reply:
The a cast right there, accompanied by an appropriate comment, should be the right thing to do.

Reply:
ok - there will be plenty of times where we want to convert between `IArray` an `Array` in the later commits - I thought this import makes it more standardised rather than having casts everywhere

Reply:
got rid of the implicit stuff and renamed to `unsafe.asByteArray`